source/PlayerInfo.cpp

```diff
@@ -2550,14 +2550,13 @@ void PlayerInfo::ValidateLoad()
 
 	// Validate the missions that were loaded. Active-but-invalid missions are removed from
 	// the standard mission list, effectively pausing them until necessary data is restored.
-	missions.sort([](const Mission &lhs, const Mission &rhs) noexcept -> bool { return lhs.IsValid(); });
-	auto isInvalidMission = [](const Mission &m) noexcept -> bool { return !m.IsValid(); };
-	auto mit = find_if(missions.begin(), missions.end(), isInvalidMission);
+	auto mit = partition(missions.begin(), missions.end(), [](const Mission &m) { return m.IsValid(); });
 	if(mit != missions.end())
 		inactiveMissions.splice(inactiveMissions.end(), missions, mit, missions.end());
 
 	// Invalid available jobs or missions are erased (since there is no guarantee
 	// the player will be on the correct planet when a plugin is re-added).
+	auto isInvalidMission = [](const Mission& m) noexcept -> bool { return !m.IsValid(); };
```

Comment:
What are we solving by changing this?

We are still sorting the list into two groups and consuming both parts, but now we use twice as many predicates, don't have stable ordering (fixable, yes), and may lose any specializations that `list::sort` can utilize that its iterators cannot.